shared/modules/factory.go

```diff
@@ -4,25 +4,31 @@ package modules
 // a variadic `ModuleOption` argument(s) and returns a `Module`and an error.
 type ModuleFactoryWithOptions FactoryWithOptions[Module, ModuleOption]
 
-// FactoryWithConfig implements a `#Create()` factory method which takes a
-// required "config" argument of type K and returns a value of type T and an error.
+// Factory implements a `#Create()` factory method which takes a bus and returns
+// a value of type T and an error.
+type Factory[T any] interface {
+	Create(bus Bus) (T, error)
+}
+
+// FactoryWithConfig implements a `#Create()` factory method which takes a bus and
```

Comment:
NIT: Thoughts on `s/T/M/g` and `s/K/C/g/` to represent `M`odule and `C`onfig respectively

Reply:
I'm in.